test/e2e/federated-service.go

```diff
@@ -110,6 +110,28 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 				}()
 				waitForServiceShardsOrFail(nsName, service, clusters)
 			})
+
+			It("should not be deleted from underlying clusters when it is deleted", func() {
+				framework.SkipUnlessFederated(f.ClientSet)
+				nsName = f.FederationNamespace.Name
+				service = createServiceOrFail(f.FederationClientset_1_5, nsName, FederatedServiceName)
+				By(fmt.Sprintf("Successfully created federated service %q in namespace %q. Waiting for shards to appear in underlying clusters", service.Name, nsName))
+
+				waitForServiceShardsOrFail(nsName, service, clusters)
+
+				// Cleanup
+				By(fmt.Sprintf("Deleting service %s", service.Name))
+				err := f.FederationClientset_1_5.Services(nsName).Delete(service.Name, &v1.DeleteOptions{})
+				framework.ExpectNoError(err, "Error deleting service %q in namespace %q", service.Name, service.Namespace)
+				By(fmt.Sprintf("Deletion of service %q in namespace %q succeeded.", service.Name, nsName))
+				By(fmt.Sprintf("Verifying that services in underlying clusters are not deleted"))
+				for clusterName, clusterClientset := range clusters {
+					_, err := clusterClientset.Core().Services(service.Namespace).Get(service.Name)
+					if err != nil {
+						framework.Failf("Unexpected error in fetching service %s in cluster %s, %s", service.Name, clusterName, err)
```

Comment:
I am not entirely sure about failing here. It should log an error and continue?

Also, I would just put this in a `defer` block.